[Ang Min Cher, Branda] iP

panpannnnn opened the pull request:
### DukePro 😁

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro is a _very simple_ task tracker. It's

- Easy to use
- Text-based
- Efficient

All you need to do is,

1. Download it from [here](https://github.com/panpannnnn/ip/releases).
2. Double-click it.
3. Learn the functions from the help menu
4. Sit back and let it manage tasks for you!

The **new and improved** DukePro allows you to,
- [X] Add tasks
- [X] Save current tasks for future use
- [X] Mark tasks as done or not done
- [x] Delete tasks
- [ ] Find tasks using keywords (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Duke {
    public static void main(String[] args) throws DukeException {
        UI ui = new UI();
        TaskList list = new TaskList(100);
        Storage storage = new Storage(list);
        Parser parser = new Parser();
        ui.start();
        storage.findData();
        storage.connect();
        start(ui, list, storage, parser);
    }
}
```

Commit:
Add Gradle support

panpannnnn committed:
Level-1

panpannnnn committed:
Level-2

panpannnnn committed:
Level-3

panpannnnn committed:
Level-4

panpannnnn committed:
A-TextUiTesting

panpannnnn committed:
Level-5

panpannnnn committed:
Level-6

panpannnnn committed:
Level-7

panpannnnn committed:
Level-8

panpannnnn committed:
A-MoreOOP

panpannnnn committed:
A-Packages

panpannnnn committed:
Merge pull request #1 from panpannnnn/add-gradle-support

Add Gradle support

kyueran commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,27 @@
+
+    @Override
+    public String toSave() {
+        if (super.isDone()) {
+            return "D | 1 | " + super.getName() + " | " + this.deadline.format(DateTimeFormatter.ofPattern("yyyy-MM-dd")) + "\n";
+        } else {
+            return "D | 0 | " + super.getName() + " | " + this.deadline.format(DateTimeFormatter.ofPattern("yyyy-MM-dd")) + "\n";
```
Can line break in the middle, before this.deadline.format

kyueran commented on src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,61 @@
+        }
+    }
+
+    public void mark(TaskList list, int index) {
+        list.getTask(index - 1).setDone();
+        System.out.println("ʕ•̀ω•́ʔ✧ :: Nice! I've marked this task as done:\n  " + list.getTask(index - 1).toString());
+    }
+
+    public void unmark(TaskList list, int index) {
+        list.getTask(index - 1).setUndone();
+        System.out.println("ʕ•̀ω•́ʔ✧ :: OK, I've marked this task as not done yet:\n  " + list.getTask(index - 1).toString());
+    }
```
Can break the line here to improve readability.

kyueran commented on src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,61 @@
+
+    public void addDeadline(TaskList list, String name, String deadline) {
+        list.addTask(new Deadline(name, deadline));
+        System.out.println("ฅʕ'ᴥ'ʔฅ :: Got it. I've added this task:");
+        System.out.println("  " + list.getTask(list.getSize() - 1).toString());
+        System.out.println("ʕ•ᴥ•ʔっ :: Now you have " + list.getSize() + " task(s) in the list.");
```
Can concatenate the strings first then call System.out.println on the overall string.

kyueran reviewed (commented):
LGTM, can consider changing some minor style changes suggested. Good job overall!

verydaftpunk commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,42 @@
+        this.done = " ";
+    }
+
+    public boolean isDone() {
```
good naming of boolean in line with coding standards

verydaftpunk commented on src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,61 @@
+        list.addTask(new Event(name, start, end));
+        System.out.println("ฅʕ'ᴥ'ʔฅ :: Got it. I've added this task:");
+        System.out.println("  " + list.getTask(list.getSize() - 1).toString());
+        System.out.println("ʕ•ᴥ•ʔっ :: Now you have " + list.getSize() + " task(s) in the list.");
```
good addition of emoticons to inject personality into duke. very good.

verydaftpunk commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,84 @@
+                    }
+                } else if (cmd.startsWith("delete")) {
+                    if (list.getSize() == 0) {
+                        throw new DukeException("ʕ ﾟ ● ﾟʔ :: ☹ OOPS!!! The list is empty!");
```
good throwing of exception if trying to delete an empty list

verydaftpunk commented on build.gradle:
```diff
@@ -0,0 +1,41 @@
+plugins {
```
your stuff is very good!

panpannnnn committed:
A-Gradle

panpannnnn committed:
A-JUnit

panpannnnn committed:
A-Jar

panpannnnn committed:
Incorrectly reversed hunk

panpannnnn committed:
A-JavaDoc

panpannnnn committed:
Edit according to coding standards.

Storage: change boolean naming to convention

Duke: improve SLAP of the start() method
Method was too long. Hence the implementation was split into different smaller methods.

panpannnnn committed:
Level-9

Create find functionality
Duke: create a new case for finding a task
Parser: create a new method to find keyword of the task
Tasklist: create new method to gather matching tasks
UI: create a new method to display matching tasks

panpannnnn committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Edit according to coding standards.

panpannnnn committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/UI.java

panpannnnn committed:
A-CheckStyle
Add checkstyle

panpannnnn committed:
Revamp duke so that ui, list, storage and parser won't have to be passed in through the methods.

panpannnnn committed:
Level-10

Add GUI to duke
Add a 'help' menu
Add a 'retrieve data' feature for users using the GUI to load previous tasks

panpannnnn committed:
A-Assertions

Add assert feature to tasklist methods to document assumptions that should hold.

panpannnnn committed:
A-Assertions

Add assert feature to tasklist methods to document assumptions that should hold.

panpannnnn committed:
A-CodeQuality

Add JavaDocs for each method and code comments.
Check for readability.
Consolidate duplicate conditional fragments for Duke.java

panpannnnn committed:
Merge pull request #3 from panpannnnn/branch-A-Assertions

Branch-A-Assertions

panpannnnn committed:
Merge pull request #4 from panpannnnn/branch-A-CodeQuality

Branch-A-CodeQuality

panpannnnn committed:
BCD-Extension

Add the C-Help extension. Actually added in an earlier commit: see commit 5f1c137

panpannnnn committed:
Add credit for GUI design

panpannnnn committed:
A-Personality

Change Duke to panpan (my own nickname:P)

panpannnnn committed:
A-UserGuide

Update README.md with the new user guide

panpannnnn committed:
Rename archive base name to 'panpan' for release

panpannnnn committed:
Update mainClassName for new release

panpannnnn committed:
Update Storage handling of file not found exception
